lib/Dialects/LinalgExt/Transforms/Tiling.cpp

```diff
@@ -32,39 +32,47 @@ SmallVector<Value> tileToSCF(PatternRewriter &rewriter, TilingInterface op,
                              TilingInterface clonedOp, ValueRange tileSizes) {
   // Compute lower and upper bounds of the loop nest.
   SmallVector<Range> ranges = clonedOp.getIterationDomain(rewriter);
-  assert(tileSizes.size() == ranges.size() &&
+  assert(tileSizes.size() <= ranges.size() &&
          "expected tile sizes to match the number of loops");
+
+  // Fill the tile sizes with zeros for the untiled dimensions.
```

Comment:
Should the "completion mechanism" be part of the interface itself?

Reply:
I am back and forth on this. Its a bit opinionated to do that, and actually the completion mechanism is different if it is a tensor semantics op or a memref semantics op. I am waiting to get enough signal to go one way or the other. What I more thinking of is instead of the completion mechanism being part of the interface, we can break it up to have an op publish tile size and offsets of the result the tiled computation updates (using the offsets and sizes specified through the interface). That might be enough to move the completion mechanism to the tiling algorithm. I add actually had that as part of this PR since I thought I needed it, but turns out I didnt.